sdk/src/main/java/io/dapr/client/DaprClientHttp.java

```diff
@@ -768,15 +772,140 @@ private <T> QueryStateResponse<T> buildQueryStateResponse(DaprHttp.Response resp
    */
   @Override
   public Mono<Map<String, ConfigurationItem>> getConfiguration(GetConfigurationRequest request) {
-    return DaprException.wrapMono(new UnsupportedOperationException());
+    try {
+      final String configurationStoreName = request.getStoreName();
+      final List<String> keys = request.getKeys();
+      if ((configurationStoreName == null) || (configurationStoreName.trim().isEmpty())) {
+        throw new IllegalArgumentException("Configuration Store Name cannot be null or empty.");
+      }
+      if (keys.isEmpty()) {
+        throw new IllegalArgumentException("Keys can not be empty or null");
+      }
+
+      String[] pathSegments = new String[] {DaprHttp.ALPHA_1_API_VERSION, "configuration", configurationStoreName };
+      Map<String, List<String>> queryParams = new HashMap<>();
+      queryParams.put("key", Collections.unmodifiableList(keys));
```

Comment:
@pravinpushkar Metadata is got from query args as well ... so that needs to be populated here ...
See https://github.com/dapr/dapr/blob/master/pkg/http/api.go#L1036